### PR TITLE
Refactor pipeline

### PR DESCRIPTION
**Pull request by star-e**
Fix shading scale error when using multiple-camera with enabled/disabled shading scale.
Add `scene` parameter to `addScene` method for reflection probe.
Adding reflection probe implementation.

### Changelog

* `addScene` interface changed. Added scene parameter.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

## Greptile Summary

Here are the key bullet points summarizing the changes in this pull request:

- Added null check for scene when getting main light in `NativeSetter.cpp`
- Modified camera scene check condition in `NativeSceneCulling.cpp`
- Improved robustness of `setBuiltinCameraConstants` method
- Updated `batchFrustumCulling` to handle cameras with null scenes

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

File: native/cocos/renderer/pipeline/custom/NativeSceneCulling.cpp [hunk 1]
```diff
@@ -312,7 +312,7 @@ void SceneCulling::batchFrustumCulling(const NativePipeline& ppl) {
         CC_ENSURES(scene);
         for (const auto& [key, frustomCulledResultID] : queries.resultIndex) {
             CC_EXPECTS(key.camera);
-            CC_EXPECTS(key.camera->getScene() == scene);
+            CC_EXPECTS(key.camera->getScene() == nullptr || key.camera->getScene() == scene);
```
[the remark this answers is not in the document]
Comment (star-e) [reply native/cocos/renderer/pipeline/custom/NativeSceneCulling.cpp:315]: null scene is allowed for reflection probe camera.

File: native/cocos/renderer/pipeline/custom/NativeSetter.cpp [hunk 2]
```diff
@@ -116,11 +116,12 @@ void NativeSetter::setMat4ArrayElem(const ccstd::string &name, const cc::Mat4 &m
 
 void NativeSetter::setBuiltinCameraConstants(const scene::Camera *camera) {
     auto &data = get(RenderGraph::DataTag{}, *renderGraph, nodeID);
+    const auto* scene = camera->getScene();
     setCameraUBOValues(
         *camera,
         *layoutGraph,
         *pipelineRuntime->getPipelineSceneData(),
-        camera->getScene()->getMainLight(), data);
+        scene ? scene->getMainLight() : nullptr, data);
```
[the remark this answers is not in the document]
Comment (star-e) [reply native/cocos/renderer/pipeline/custom/NativeSetter.cpp:124]: ~~null scene is already checked when getting main light.~~

**Comment by star-e** [comment]
@cocos-robot run test cases

**Review by minggo — APPROVED** [review]

**Pull request merged**